### ISSUE TITLE
Implement ONLINE and OTHER search criteria for locations.

### ISSUE THREAD

**Issue by courtney-rosenthal** [courtney-rosenthal/apl-event-finder#10]
Currently, there is an OTHER location in CalendarEventSearchCriteria.Location, but it is not being assigned to events. Instead, the CalendarEvent.Location.key fields is being set to null.

There is not a location for online events.

Under this ticket:
- Change Location.key from nullable to non-nullable
- Assign the location for "Other" rather than null to locations that don't otherwise match.
- Create a new location for ONLINE, and assign that (instead of OTHER) for events whose location is online.
- Be sure to there are test cases for ONLINE and OTHER events.

This work is entirely in the _lib-elasticsearch_ module.

The work on modifying the search for these new criteria will be a follow-on issues.

**Comment by courtney-rosenthal** [comment]
It's possible that search as currently implemented will just work after this change. Should verify that once this work is complete, and open new issue as needed.
